crates/nu-command/src/platform/input/input_.rs

```diff
@@ -59,114 +68,104 @@ impl Command for Input {
-                            KeyCode::Char(c) => {
-                                if k.modifiers == KeyModifiers::ALT
-                                    || k.modifiers == KeyModifiers::CONTROL
-                                {
-                                    if k.modifiers == KeyModifiers::CONTROL && c == 'c' {
-                                        crossterm::terminal::disable_raw_mode()
-                                            .map_err(&from_io_error)?;
-                                        return Err(IoError::new(
-                                            std::io::ErrorKind::Interrupted,
-                                            call.head,
-                                            None,
-                                        )
-                                        .into());
-                                    }
-                                    continue;
-                                }
-
-                                if let Some(bytes_until) = bytes_until.as_ref() {
-                                    if bytes_until.bytes().contains(&(c as u8)) {
-                                        break;
-                                    }
-                                }
-                                buf.push(c);
-                            }
-                            KeyCode::Backspace => {
-                                let _ = buf.pop();
-                            }
-                            KeyCode::Enter => {
-                                break;
-                            }
-                            _ => continue,
-                        }
-                    }
-                    _ => continue,
-                },
-                Ok(_) => continue,
-                Err(event_error) => {
-                    crossterm::terminal::disable_raw_mode().map_err(&from_io_error)?;
-                    return Err(from_io_error(event_error).into());
-                }
+            Ok(Signal::CtrlC) => {
+                return Err(IoError::new(std::io::ErrorKind::Interrupted, call.head, None).into());
             }
-            if !suppress_output {
-                // clear the current line and print the current buffer
-                execute!(
-                    std::io::stdout(),
-                    terminal::Clear(ClearType::CurrentLine),
-                    cursor::MoveToColumn(0),
-                )
-                .map_err(|err| IoError::new(err.kind(), call.head, None))?;
-                if let Some(prompt) = &prompt {
-                    execute!(std::io::stdout(), Print(prompt.to_string()))
-                        .map_err(&from_io_error)?;
-                }
-                execute!(std::io::stdout(), Print(buf.to_string())).map_err(&from_io_error)?;
+            Ok(Signal::CtrlD) => {
+                // Do nothing on ctrl-d
+                return Ok(Value::nothing(call.head).into_pipeline_data());
```

Comment:
Should this raise another error? We would need to document that this has a different behavior than ctrl-c

Reply:
repl I tried don't raise an error on ctrl+d, 
exiting a nu subshell don't error either.

this is where the inspiration came from

<img width="490" alt="image" src="https://github.com/user-attachments/assets/4c224c6b-6ad5-4700-a8a7-86a0189fe10b" />

Reply:
(happy to change it!)

Reply:
In a REPL the convention is `Ctrl-C` interrupt the current entry but continue the loop and `Ctrl-D` end the whole repl session (after a delete overload in emacs/bash style bindings). 
But `input` itself doesn't implement a whole repl but a singular input, if you want to use it for that you would want to have handle that in your repl application written in nu.
I personally would find the return value nothing surprising if I expect a string. For Ctrl-C we seem to take general precendence and block out the direct handling by a script as it is our general interrupt for scripts.

Not sure what the best way to signal it to the `input` users that build their scripts/modules is. The dreaded `null` or an error they would have to `try`/`catch`.